src/main/java/seedu/address/model/lesson/TimeRange.java

```diff
@@ -3,77 +3,83 @@
-     * @param end End of the range.
+     * @param value A valid time range.
      */
-    public TimeRange(Time start, Time end) {
-        requireNonNull(start);
-        requireNonNull(end);
-        this.start = start;
-        this.end = end;
-        checkArgument(isValidTimeRange(start, end), MESSAGE_CONSTRAINTS);
+    public TimeRange(String value) {
+        requireNonNull(value);
+        checkArgument(isValidTimeRange(value), MESSAGE_CONSTRAINTS);
+        this.value = value;
+        String[] startEndTimes = value.split("-", 2);
+        start = LocalTime.parse(startEndTimes[0], DATE_TIME_FORMAT);
+        end = LocalTime.parse(startEndTimes[1], DATE_TIME_FORMAT);
     }
 
-    public Time getStart() {
+    public LocalTime getStart() {
         return start;
     }
 
-    public Time getEnd() {
+    public LocalTime getEnd() {
         return end;
     }
 
+    public Duration getDuration() {
+        return Duration.between(start, end);
+    }
+
     /**
      * Checks if the start is earlier than the end and
      * the range is between 8am and 10pm, inclusive.
      *
-     * @param start Start of the range.
-     * @param end End of the range.
+     * @param test The value to be tested.
      * @return True if start is earlier than or same time as end.
      */
-    public static boolean isValidTimeRange(Time start, Time end) {
-        return end.compareTo(start) >= 0
-            && start.compareTo(DAY_START) >= 0 // Same or later than 8am
-            && end.compareTo(DAY_END) <= 0; // Same or earlier than 10pm
-    }
-
-    /**
-     * Checks if the given start time is within the time range.
-     *
-     * @param time The time to be tested.
-     */
-    private boolean isStartClashing(Time time) {
-        return start.compareTo(time) <= 0 // same or later than start time
-            && end.compareTo(time) > 0; // earlier than end time
-    }
-
-    /**
-     * Checks if the given end time is within the time range.
-     *
-     * @param time The time to be tested.
-     */
-    private boolean isEndClashing(Time time) {
-        return start.compareTo(time) < 0 // later than start time
-            && end.compareTo(time) > 0; // earlier than end time
+    public static boolean isValidTimeRange(String test) {
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
+        }
+        String[] startEndTimes = test.split("-", 2);
+        if (startEndTimes.length < 2) { // There must be a '-'
+            return false;
+        }
+        try {
+            LocalTime startTime = LocalTime.parse(startEndTimes[0], DATE_TIME_FORMAT);
+            LocalTime endTime = LocalTime.parse(startEndTimes[1], DATE_TIME_FORMAT);
+            return endTime.compareTo(startTime) > 0 // End cannot be same or before start
+                    && startTime.compareTo(DAY_START) >= 0 // Same or later than 8am
+                    && endTime.compareTo(DAY_END) <= 0; // Same or earlier than 10pm
```

Comment:
same thing hehe i think can make it `isValidStartEndTime`, `isValidStartTime` and `isValidEndTime`